api/proto/config/appconfig.proto

```diff
@@ -26,6 +26,47 @@ enum MetaDataType {
   MetaDataDriveMultipart = 3; // Process multipart MIME for application
 }
 
+// The information about a snapshot
+message AvailableSnapshot {
+  string id = 1;
+}
+
+// The snapshot information for an Application Instance
+message SnapshotConfig {
+  // activeSnapshot is the id of the snapshot which is currently used.
+  // It may be empty if no snapshot is used. Worth to note that the field is
+  // not used alone to trigger a rollback. It just indicates that the device
+  // is currently using a snapshot with the given id. To trigger a rollback, in
+  // addition to it, the rollbackCounter should be increased. It is done to
+  // avoid unnecessary rollbacks when the same activeSnapshot is sent several
+  // times (we cannot relay on EVE to perform rollbacks after the first time
+  // it receives the corresponding message, so it's a regular case to send the
+  // same value several times). Also it is used to perform a rollback to the
+  // same snapshot several times.
+  string activeSnapshot = 1;
+  // rollbackCounter is necessary in addition to the activeSnapshot to request
+  // EVE to trigger a rollback. The rollbackCounter is incremented when a
+  // snapshot is used for a rollback. It should not be decreased.
+  uint32 rollbackCounter = 2;
+  // maxSnapshots is the maximum number of snapshots that can be stored for the
+  // application instance.
+  // If the number of snapshots exceeds the maxSnapshots, the oldest snapshot
+  // will be deleted.
```

Comment:
What if the oldest snapshot is currently the active one? Does it still get deleted?

Reply:
Yes, I think so.
Let's consider the following scenario:
We have 1 as the value of the max snapshots stored.
We make a snapshot.
VM continues working.
We roll back to the snapshot created earlier (so the first snapshot becomes active).
EVE performs the rollback.
We request a new snapshot be created.
Yes, I think the first one should be deleted in this case.

Reply:
Please update comment in proto file to make this explicit.

Reply:
Done